ytypes/leaf.go

```diff
@@ -508,117 +509,203 @@ the resulting Bgp_Neighbor_RouteReflector would have field
 RouteReflectorClusterId set with the type Bgp_Neighbor_RouteReflector_RouteReflectorClusterId_Union_String,
 with field String set to "forty-two".
 */
+
 func unmarshalUnion(schema *yang.Entry, parent interface{}, fieldName string, value interface{}) error {
 	util.DbgPrint("unmarshalUnion value %v, type %T, into parent type %T field name %s, schema name %s", util.ValueStr(value), value, parent, fieldName, schema.Name)
-	v, t := reflect.ValueOf(parent), reflect.TypeOf(parent)
-	if !util.IsTypeStructPtr(t) {
+	parentV, parentT := reflect.ValueOf(parent), reflect.TypeOf(parent)
+	if !util.IsTypeStructPtr(parentT) {
 		return fmt.Errorf("%T is not a struct ptr in unmarshalUnion", parent)
 	}
 
-	vf := v.Elem().FieldByName(fieldName)
-	if !vf.IsValid() {
+	// Get the value and type of the field to set, which may have slice or
+	// interface types for leaf-list and union cases.
+	destUnionFieldV := parentV.Elem().FieldByName(fieldName)
+	if !destUnionFieldV.IsValid() {
 		return fmt.Errorf("%s is not a valid field name in %T", fieldName, parent)
 	}
-	ft, err := getFieldElemType(parent, fieldName)
+	dft, _ := parentT.Elem().FieldByName(fieldName)
```

Comment:
What is the `_` here? AFAICS `FieldByName` only returns one `reflect.Value`?

Reply:
This is the method for the Type (which doesn't come up in godoc, seems like a rendering bug):
// FieldByName returns the struct field with the given name
        // and a boolean indicating if the field was found.
        FieldByName(name string) (StructField, bool)
We don't need to check ok because we known it exists from the line above.

Reply:
Ah, thanks -- strange, I looked a GoDoc, weird that it has a bug here.